modules/core/include/opencv2/core/hal/intrin_avx512.hpp

```diff
@@ -3067,6 +3067,61 @@ inline int v_scan_forward(const v_int64x8& a) { return trailingZeros32(v_signmas
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 10) + pos) + channel]), 10);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 11) + pos) + channel]), 11);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 12) + pos) + channel]), 12);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 13) + pos) + channel]), 13);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 14) + pos) + channel]), 14);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 15) + pos) + channel]), 15);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 16) + pos) + channel]), 16);
+
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 17) + pos) + channel]), 17);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 18) + pos) + channel]), 18);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 19) + pos) + channel]), 19);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 20) + pos) + channel]), 20);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 21) + pos) + channel]), 21);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 22) + pos) + channel]), 22);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 23) + pos) + channel]), 23);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 24) + pos) + channel]), 24);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 25) + pos) + channel]), 25);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 26) + pos) + channel]), 26);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 27) + pos) + channel]), 27);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 28) + pos) + channel]), 28);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 29) + pos) + channel]), 29);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 30) + pos) + channel]), 30);
+        vec.val = v512_insert_epi16(vec.val, *reinterpret_cast<const uchar*>(&src[chanNum * (*(index + 31) + pos) + channel]), 31);
```

Comment:
Could it be a loop?

Reply:
It's not our code actually

Reply:
Why is this code in diff?

Reply:
Or maybe it is. If so, please, combine such lines in loops where applicable